axum-extra/src/response/erased_json.rs

```diff
@@ -72,3 +81,61 @@ impl IntoResponse for ErasedJson {
         }
     }
 }
+
+/// Construct an [`ErasedJson`] response from a JSON literal.
+///
+/// A `Content-Type: application/json` header is automatically added.
+/// Any variable or expression implementing [`Serialize`]
+/// can be interpolated as a value in the literal.
+/// If the [`Serialize`] implementation decides to fail,
+/// or if a map with non-string keys is used,
+/// a 500 response will be issued
+/// whose body is the error message in UTF-8.
+///
+/// Internally,
+/// this function uses the [`typed_json::json!`] macro
+/// so that no allocations are performed during serialization.
```

Comment:
`ErasedJson::new` *does* allocate, so this wording needs to be tweaked a bit. Maybe we could state that it allocates a lot less than `Json(serde_json::json!(...))`?